Mathlib/Algebra/Ring/SumsOfSquares.lean

```diff
@@ -16,96 +18,175 @@ instances. Sums of squares in `R` are defined by an inductive predicate `IsSumSq
 `0 : R` is a sum of squares and if `S` is a sum of squares, then for all `a : R`, `a * a + S` is a
 sum of squares in `R`.
 
-## Main declaration
+## Declarations
 
 - The predicate `IsSumSq : R → Prop`, defining the property of being a sum of squares in `R`.
-
-## Auxiliary declarations
-
-- The type `sumSqIn R` : in an additive monoid with multiplication `R`, we introduce the type
-`sumSqIn R` as the submonoid of `R` whose carrier is the subset `{S : R | IsSumSq S}`.
+- The terms `AddMonoid.sumSqIn R` and `Subsemiring.sumSqIn R` :
+in an additive monoid with multiplication `R` and a semiring `R`, we introduce the terms
+`AddMonoid.sumSqIn R` and `Subsemiring.sumSqIn R` as the submonoid and subsemiring, respectively,
+of `R` whose carrier is the subset `{S : R | IsSumSq S}`.
 
 ## Theorems
 
+- `IsSquare.isSumSq`, `IsSumSq.mul_self`: a square in `R` is a sum of squares in `R`.
 - `IsSumSq.add`: if `S1` and `S2` are sums of squares in `R`, then so is `S1 + S2`.
-- `IsSumSq.nonneg`: in a linearly ordered semiring `R` with an `[ExistsAddOfLE R]` instance, sums of
-squares are non-negative.
-- `mem_sumSqIn_of_isSquare`: a square is a sum of squares.
+- `IsSumSq.mul`: if `S1` and `S2` are sums of squares in `R`, then so is `S1 * S2`.
+- `IsSumSq.sum` : a finite sum of sums of squares in `R` is a sum of squares.
+- `IsSumSq.prod` : a finite product of sums of squares in `R` is a sum of squares.
+- `IsSumSq.nonneg`: sums of squares are non-negative.
 - `AddSubmonoid.closure_isSquare`: the submonoid of `R` generated by the squares in `R` is the set
 of sums of squares in `R`.
+- `Subsemiring.closure_isSquare`: the subsemiring of `R` generated by the squares in `R` is the set
+of sums of squares in `R`.
 
 -/
 
-variable {R : Type*} [Mul R]
+variable {R : Type*}
 
 /--
 In a type `R` with an addition, a zero element and a multiplication, the property of being a sum of
 squares is defined by an inductive predicate: `0 : R` is a sum of squares and if `S` is a sum of
 squares, then for all `a : R`, `a * a + S` is a sum of squares in `R`.
 -/
 @[mk_iff]
-inductive IsSumSq [Add R] [Zero R] : R → Prop
+inductive IsSumSq [Mul R] [Add R] [Zero R] : R → Prop
   | zero                              : IsSumSq 0
-  | sq_add (a S : R) (pS : IsSumSq S) : IsSumSq (a * a + S)
-
-@[deprecated (since := "2024-08-09")] alias isSumSq := IsSumSq
-
```

Comment:
Flo and I thought it would be fine, here
https://github.com/leanprover-community/mathlib4/pull/16094#discussion_r1729199213
Should I reinstate deprecated names for those declarations which I haven't rewritten? I can understand why that would be sensible.

Reply:
Yes, it looks like the policy is to keep the deprecations for 3-6 months and definitely at least one month so that they make it to at least one `stable` release